openapi.yaml

```diff
@@ -223,188 +291,56 @@ components:
-
-    CreateLogRequest:
-      type: object
-      properties:
-        projectName:
+        content:
           type: string
           nullable: true
-        versionName:
-          type: string
+        jsonContent:
+          type: object
           nullable: true
-        data:
+          additionalProperties: true
+        metadata:
           type: object
+          additionalProperties:
+            type: string
+          default: {}
       required:
-        - data
-
-    LoggedSample:
-      type: object
-      properties:
-        id:
-          type: string
-          description: The ID of the logged sample
-        data:
-          type: string
-          description: The log data
-        projectId:
-          type: string
-          description: The ID of the project
-        versionId:
-          type: string
-          description: The ID of the project version
-
-    InternalExperimentEggNoSamplesNoComparisons:
-      type: object
-      properties:
-        name:
-          type: string
-        projectId:
-          type: integer
-          nullable: true
-        instructions:
-          type: string
-          nullable: true
-
-    InternalExperimentEgg:
-      type: object
-      properties:
-        name:
-          type: string
-        instructions:
-          type: string
-          nullable: true
-        project:
-          type: string
-          nullable: true
-        version:
-          type: string
-          nullable: true
+        - role
+      # Validation rules matching the TypeScript refinements
       oneOf:
-        - not:
-            required:
-              - project
-        - required:
-            - project
-            - version
-
-    InternalExperimentEggWithProject:
-      type: object
-      properties:
-        name:
-          type: string
-        instructions:
-          type: string
-          nullable: true
-        project:
-          type: string
-        version:
-          type: string
-
-    InternalSampleEgg:
-      type: object
-      properties:
-        title:
-          type: string
-          nullable: true
-        message:
-          type: string
-          nullable: true
-        response:
-          type: string
-
-    InternalComparisonEgg:
-      type: object
-      properties:
-        samples:
-          type: array
-          items:
-            $ref: '#/components/schemas/InternalSampleEgg'
-
-    InternalComparisonEggWithVersion:
-      type: object
-      properties:
-        samples:
-          type: array
-          items:
-            $ref: '#/components/schemas/InternalSampleEggWithVersion'
-
-    InternalSampleEggWithVersion:
-      type: object
-      properties:
-        title:
-          type: string
-          nullable: true
-        message:
-          type: string
-          nullable: true
-        response:
-          type: string
-        version:
-          type: string
+        - properties:
```

Comment:
It looks like [there is a way to give these titles](https://mintlify.com/docs/api-playground/openapi/advanced-features#providing-options-with-oneof-and-anyof) so that we can call them something like "Markdown Message" and "Json Message"

I think that would be helpful to make this a little less confusing: 

<img width="635" alt="Screenshot 2024-10-31 at 12 05 52 PM" src="https://github.com/user-attachments/assets/c165140d-be40-4d03-aaec-77009a422152">

Reply:
Also, once I select the markdown option - ideally the "jsonContent" field wouldn't be there anymore - my guess is the best way to accomplish this is to create two new schemas for JsonMessage and MarkdownMessage and make the message definition just by "oneOf" a reference to those two schemas. Let me know if that doesn't make sense.

Here's what i see now that seems imperfect:

<img width="516" alt="Screenshot 2024-10-31 at 12 07 35 PM" src="https://github.com/user-attachments/assets/158e5f56-4474-46d0-a354-321fff813361">